File: infrastruktur/src/main/java/com/github/haschi/haushaltsbuch/infrastruktur/LoggingCallback.java
package com.github.haschi.haushaltsbuch.infrastruktur;

import io.vertx.core.logging.Logger;
import io.vertx.core.logging.LoggerFactory;
import org.axonframework.commandhandling.CommandCallback;
import org.axonframework.commandhandling.CommandMessage;

import java.text.MessageFormat;

public final class LoggingCallback implements CommandCallback<Object, Object>
{
    private final Logger logger = LoggerFactory.getLogger(LoggingCallback.class);

    public static final LoggingCallback INSTANCE = new LoggingCallback();

    private LoggingCallback()
    {
    }

    @Override
    public void onSuccess(final CommandMessage message, final Object result)
    {
        logger.info(
                MessageFormat.format(
                        "[{0}] {1} => OK {2}",
                        message.getCommandName(),
                        message.getIdentifier(),
                        result));
    }

    @Override
    public void onFailure(final CommandMessage<?> message, final Throwable cause)
    {
        logger.warn(
                MessageFormat.format(
                        "[{0}] {1} => OK {2}",
                        message.getCommandName(),
                        message.getIdentifier(),
                        cause.getLocalizedMessage()));
    }
}

File: infrastruktur/src/main/kotlin/com/github/haschi/haushaltsbuch/infrastruktur/LoggingCallback.kt
package com.github.haschi.haushaltsbuch.infrastruktur

import org.axonframework.commandhandling.CommandCallback
import org.axonframework.commandhandling.CommandMessage
import org.slf4j.LoggerFactory

class LoggingCallback private constructor() : CommandCallback<Any, Any>
{
    private val logger = LoggerFactory.getLogger("COMMAND")

    override fun onSuccess(message: CommandMessage<*>, result: Any?)
    {
        logger.info("[${message.commandName}] ${message.identifier} => OK ${result ?: ""}")
    }

    override fun onFailure(message: CommandMessage<*>, cause: Throwable)
    {
        logger.warn("[${message.commandName}] ${message.identifier} => ERR ${cause.localizedMessage}")
    }

    companion object
    {
        val INSTANCE = LoggingCallback()
    }
}
